src/eduid_common/config/idp.py

```diff
@@ -0,0 +1,190 @@
+    # URL (string) for use in simple templating of login.html.
+    signup_link: str = '#'
+    # URL (string) for use in simple templating of forbidden.html.
+    dashboard_link: str = '#'
+    # URL (string) for use in simple templating of login.html.
+    password_reset_link: str = '#'
+    # More links
+    technicians_link: str = '#'
+    student_link: str = '#'
+    staff_link: str = '#'
+    faq_link: str = '#'
+    # Default language code to use when looking for web pages ('en').
+    default_language: str = 'en'
+    # Base URL of the IdP. The default base URL is constructed from the
+    # Request URI, but for example if there is a load balancer/SSL
+    # terminator in front of the IdP it might be required to specify
+    # the URL of the service.
+    base_url: Optional[str] = None
+    # The scope to append to any unscoped eduPersonPrincipalName
+    # attributes found on users in the userdb.
+    default_eppn_scope: Optional[str] = None
+    # Disallow login for a user after N failures in a given month.
+    # This is said to be an imminent Kantara requirement.
+    # Kantara 30-day bad authn limit is 100
+    max_auhtn_failures_per_month: int = 50
+    max_authn_failures_per_month: int = 50
+    # Lifetime of state kept in IdP login phase.
+    # This is the time, in minutes, a user has to complete the login phase.
+    # After this time, login cannot complete because the SAMLRequest, RelayState
+    # and possibly other needed information will be forgotten.
+    login_state_ttl: int = 5
+    # Add a default eduPersonScopedAffiliation if none is returned from the
+    # attribute manager.
+    default_scoped_affiliation: Optional[str] = None
+    # URL to use with VCCS client. BCP is to have an nginx or similar on
+    # localhost that will proxy requests to a currently available backend
+    # using TLS.
+    vccs_url: str = 'http://localhost:8550/'
+    insecure_cookies: bool = False
+    # URI of the actions app.
+    actions_app_uri: Optional[str] = 'http://actions.example.com/'
+    # The plugins for pre-authentication actions that need to be loaded
+    action_plugins: List[str] = field(default_factory=list)
+    # The current version of the terms of use agreement.
+    tou_version: str = 'version1'
+    # The interval which a user needs to reaccept an already accepted ToU (in seconds)
+    tou_reaccept_interval: int = 94608000
+    # Name of cookie used to persist session information in the users browser.
+    shared_session_cookie_name: str = 'sessid'
+    # Key to decrypt shared sessions.
+    shared_session_secret_key: Optional[str] = None
+    # TTL for shared sessions.
+    shared_session_ttl: int = 300
+    http_headers: str = "Content-Security-Policy:default-src 'self'; script-src 'self' 'unsafe-inline', X-Frame-Options:DENY"
+    privacy_link: str = "http://html.eduid.docker/privacy.html"
+
+
+def init_config(test_config: Optional[dict] = None, debug: bool = True) -> IdPConfig:
```

Comment:
Wouldn't if be nice if this was a class method?

Can debug please default to False.

Reply:
yes good idea :)